source_gen/lib/builder.dart

```diff
@@ -41,6 +46,43 @@ Builder combiningBuilder([BuilderOptions options = BuilderOptions.empty]) {
   return builder;
 }
 
+Map<String, List<String>> _validatedBuildExtensionsFrom(
+    Map<String, dynamic> optionsMap) {
+  final extensionsOption = optionsMap.remove('build_extensions');
+  if (extensionsOption == null) return _defaultExtensions;
+
+  if (extensionsOption is! Map) {
+    throw ArgumentError(
+        'Configured build_extensions should be a map from inputs to outputs.');
+  }
+
+  final result = <String, List<String>>{};
+
+  for (final entry in extensionsOption.entries) {
+    final input = entry.key;
+    if (input is! String || !input.endsWith('.dart')) {
+      throw ArgumentError('Invalid key in build_extensions option: `$input` '
+          'should be a string ending with `.dart`');
+    }
+
+    final output = entry.value;
+    if (output is! String || !output.endsWith('.dart')) {
+      throw ArgumentError('Invalid output extension `$output`. It should be a '
+          'string ending with `.dart`');
+    }
+
+    result[input] = [output];
+  }
+
+  if (result.isEmpty) {
+    log.warning('Using default extensions as all values from the '
```

Comment:
I think this should probably throw as well - I think this is only reachable if somebody provides an empty config, which should probably just be an error.